zilencer/management/commands/populate_db.py

```diff
@@ -232,31 +236,64 @@ def handle(self, **options: Any) -> None:
             # Create subscriptions to streams.  The following
             # algorithm will give each of the users a different but
             # deterministic subset of the streams (given a fixed list
-            # of users).
+            # of users). For the test suite, we have a fixed list of
+            # subscriptions to make sure test data is consistent
+            # across platforms.
             subscriptions_to_add = []  # type: List[Subscription]
             event_time = timezone_now()
             all_subscription_logs = []  # type: (List[RealmAuditLog])
-            profiles = UserProfile.objects.select_related().filter(
-                is_bot=False, is_guest=False).order_by("email")  # type: Sequence[UserProfile]
-            for i, profile in enumerate(profiles):
-                # Subscribe to some streams.
-                for type_id in recipient_streams[:int(len(recipient_streams) *
-                                                      float(i)/len(profiles)) + 1]:
-                    r = Recipient.objects.get(type=Recipient.STREAM, type_id=type_id)
-                    s = Subscription(
-                        recipient=r,
-                        user_profile=profile,
-                        color=STREAM_ASSIGNMENT_COLORS[i % len(STREAM_ASSIGNMENT_COLORS)])
-
-                    subscriptions_to_add.append(s)
-
-                    log = RealmAuditLog(realm=profile.realm,
-                                        modified_user=profile,
-                                        modified_stream_id=type_id,
-                                        event_last_message_id=0,
-                                        event_type='subscription_created',
-                                        event_time=event_time)
-                    all_subscription_logs.append(log)
+            if options["test_suite"]:
+                subscriptions_map = {
+                    'aaron': ['Verona'],
+                    'Cordelia Lear': ['Verona'],
+                    'King Hamlet': ['Verona', 'Denmark'],
+                    'Iago': ['Verona', 'Denmark', 'Scotland'],
+                    'Othello, the Moor of Venice': ['Verona', 'Denmark', 'Scotland'],
+                    'Prospero from The Tempest': ['Verona', 'Denmark', 'Scotland', 'Venice'],
+                    'Zoe': ['Verona', 'Denmark', 'Scotland', 'Venice', 'Rome']
+                }
```

Comment:
I would have the keys be emails, not names; those are a more proper unique key in our codebase.